lib/puppet/pops/types/type_calculator.rb

```diff
@@ -452,9 +452,18 @@ def instance_of_PTupleType(t, o)
 
   def instance_of_PStructType(t, o)
     return false unless o.is_a?(Hash)
-    h = t.hashed_elements
-    # all keys must be present and have a value (even if nil/undef)
-    (o.keys - h.keys).empty? && h.all? { |k,v| instance_of(v, o[k]) }
```

Comment:
oh, so this was an optimization - not a fix. I am fine with that, but put it in a separate commit please.